### ISSUE TITLE
Bad constructor

### ISSUE THREAD

**Issue by marekfoltyn** [DuffsDevice/tinyutf8#3]
Hi,
I have found very interesting bug. When working with utf8_string with length 25, the string is bugged:

```cpp
utf8_string s = "aaaaaaaaaaaaaaaaaaaaaaaaa";
utf8_string s1(s.cpp_str());

std::cout << s.cpp_str() << std::endl;
std::cout << s1.cpp_str() << std::endl;
```

This code has following output:

```
aaaaaaaaaaaaaaaaaaaaaaaaa
aaaaaaaaaaaaaaaaaaaaaaaaІ�\x03\x01
```

The same error is when using copy constructor `utf8_string s1(s);`. 
Any idea, what could be wrong? Thank you.

**Comment by DuffsDevice** [comment]
Thank you for pointing me to it. It could be, that I my own copy which I currently work at and test with already has this bug removed. I'll check that as soon as I get to. If not, I'll go and fix it. Thank you!

**Comment by DuffsDevice** [comment]
What Compiler and Operating System do you have? I don't seem to be able to reproduce the bug.
I am on 32Bit "g++ (i686-posix-sjlj-rev1, Built by MinGW-W64 project) 6.2.0" on Windows

**Comment by marekfoltyn** [comment]
macOS, compiler: Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.4.0

**Comment by DuffsDevice** [comment]
Unfortunately, I don't have an apple pc available for testing, is there a way you can debug the code?
I have tested now with both x86_64 and x86 and couldn't reproduce it. Also, is the bug now fixed with Release 2.0? I am sorry for not being able to help you more at the moment.

**Comment by marekfoltyn** [comment]
After updating to newest commit, the issue appears to be solved. Thanks!
